components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -1064,3 +1066,114 @@ func (backend *ESClient) SetDailyLatestToFalseForControlIndex(ctx context.Contex
+func (backend *ESClient) GetNodesDayLatestTrue(ctx context.Context, time90daysAgo time.Time) ([]relaxting.NodesUpgradation, error) {
+	var nodes []relaxting.NodesUpgradation
+	indices, err := GetIndexForARange(time90daysAgo, time.Now())
+	if err != nil {
+		return nil, err
+	}
+
+	indicesSlice := strings.Split(indices, ",")
+
+	boolQuery := elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("day_latest", true)).
+		Must(elastic.NewTermQuery("daily_latest", true))
+
+	fsc := elastic.NewFetchSourceContext(true).Include(
+		"node_uuid",
+		"end_time",
+		"day_latest",
+	)
+
+	searchSource := elastic.NewSearchSource().
+		FetchSourceContext(fsc).
+		Query(boolQuery).
+		Size(10000).Sort("end_time", false)
+
+	for _, indx := range indicesSlice {
+		searchResult, err := backend.client.Search().
+			SearchSource(searchSource).
+			Index(indx).
+			FilterPath(
+				"took",
+				"hits.total",
+				"hits.hits._id",
+				"hits.hits._source",
+				"hits.hits.inner_hits").
+			Do(ctx)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if searchResult.TotalHits() > 0 {
+			// Iterate through results
+			for _, hit := range searchResult.Hits.Hits {
+				var node relaxting.NodesUpgradation
+				if hit.Source != nil {
+					err := json.Unmarshal(hit.Source, &node)
+					if err != nil {
+						logrus.Errorf("Received error while unmarshling %+v", err)
+					}
+
+				}
+				nodes = append(nodes, node)
+			}
+		}
+	}
+
+	return nodes, nil
+}
+
+//SetNodesDayLatestFalse Sets the flag for the currently present data in os database
+func (backend *ESClient) SetNodesDayLatestFalse(ctx context.Context) error {
+	bulkRequest := backend.client.Bulk()
+	script := elastic.NewScript("ctx._source.day_latest = false")
+	time90DaysAgo := time.Now().Add(-24 * time.Hour * 90)
+	nodes, err := backend.GetNodesDayLatestTrue(ctx, time90DaysAgo)
+	if err != nil {
+		return nil
+	}
+	for _, node := range nodes {
+		nodeEndTime, err := time.Parse(time.RFC3339, node.EndTime)
+		if err != nil {
+			logrus.Error("cannot parse: ", err)
+		}
+		indices, err := GetIndexForARange(time90DaysAgo, nodeEndTime.Add(-24*time.Hour))
+		if err != nil {
+			logrus.Error("cannot get indices:", err)
+		}
+
+		bulkRequest = bulkRequest.Add(elastic.NewBulkUpdateRequest().
+			Index(indices).
+			Id(node.NodeUUID).
+			Script(script))
+	}
+	approxBytes := bulkRequest.EstimatedSizeInBytes()
+	bulkResponse, err := bulkRequest.Refresh("false").Do(ctx)
+	if err != nil {
+		logrus.Errorf("Unable to send the request in bulk: %v", err)
+		return err
+	}
+	if bulkResponse == nil {
+		logrus.Errorf("Unable to fetch the response of bulk request: %v", err)
+		return err
+	}
+	logrus.Debugf("Bulk insert %d summaries, ~size %dB, took %dms", len(nodes), approxBytes, bulkResponse.Took)
+	return nil
+}
+
+//GetIndexForARange Get the index for the entire range when the start date and end date are passed from os
+func GetIndexForARange(startTime, endTime time.Time) (string, error) {
```

Comment:
still, I don't see any unit test for this method.

Reply:
Working on the test cases.